src/components/MessageList/VirtualizedMessageList.js

```diff
@@ -0,0 +1,221 @@
+import React, {
+  useContext,
+  useState,
+  useEffect,
+  useCallback,
+  useRef,
+} from 'react';
+import PropTypes from 'prop-types';
+import { Virtuoso } from 'react-virtuoso';
+
+import MessageNotification from './MessageNotification';
+import { ChannelContext, TranslationContext } from '../../context';
+import { FixedHeightMessage } from '../Message';
+import { EventComponent } from '../EventComponent';
+import { LoadingIndicator as DefaultLoadingIndicator } from '../Loading';
+import { EmptyStateIndicator as DefaultEmptyStateIndicator } from '../EmptyStateIndicator';
+
+/**
+ * VirtualizedMessageList - This component renders a list of messages in a virtual list. Its a consumer of [Channel Context](https://getstream.github.io/stream-chat-react/#channel)
+ * It is pretty fast for rendering thousands of messages but it needs its Message componet to have fixed height
+ * @example ../../docs/VirtualizedMessageList.md
+ */
+const VirtualizedMessageList = ({
```

Comment:
@mahboubii typescript?

Reply:
to soon, will add typescript and some tests once we have some feedbacks